llvm-spirv/lib/SPIRV/SPIRVLowerMemmove.cpp

```diff
@@ -75,10 +75,12 @@ class SPIRVLowerMemmove : public ModulePass,
       report_fatal_error("llvm.memmove of non-constant length not supported",
                          false);
     auto *Length = cast<ConstantInt>(I.getLength());
-    if (isa<BitCastInst>(Src))
-      // The source could be bit-cast from another type,
-      // need the original type for the allocation of the temporary variable
-      SrcTy = cast<BitCastInst>(Src)->getOperand(0)->getType();
+    auto *S = Src;
+    // The source could be bit-cast or addrspacecast from another type,
+    // need the original type for the allocation of the temporary variable
+    while (isa<BitCastInst>(S) || isa<AddrSpaceCastInst>(S))
```

Comment:
Should we use https://github.com/intel/llvm/blob/803f6cc552e8b557d05958a1aa06e44bad0294ae/llvm/lib/Analysis/ValueTracking.cpp#L4082 instead?

Reply:
This is cherry-pick of https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/7df22691c96bacfe3daee501754f253745c45602.

Reply:
I think we should avoid unnecessary changes here and this to https://github.com/KhronosGroup/SPIRV-LLVM-Translator.